src/app/online-tracker/online-tracker.component.ts

```diff
@@ -1,46 +1,14 @@
 import { Component, OnInit } from '@angular/core';
-import { Observable } from 'rxjs/Observable';
-import { Http, Response, Headers } from '@angular/http';
-
-import { SessionService } from '../openmrs-api/session.service';
+import { OnlineTrackerService } from './online-tracker.service';
 
 @Component({
   selector: 'online-tracker',
-  template: `
-    <p *ngIf="isOnline" class="text-bold"><i class="fa fa-circle text-success"></i>
-      <span></span>
-    </p>
-    <p *ngIf="!isOnline" class="text-bold"><i class="fa fa-circle text-danger"></i>
-      <span class="text-danger"><span *ngIf="isUpdating"> (updating...) </span></span>
-    </p>
-  `
+  templateUrl: './online-tracker.component.html'
 })
-export class OnlineTrackerComponent implements OnInit {
-  public isOnline: boolean = false;
-  public isUpdating: boolean = false;
-
-  constructor(private sessionService: SessionService) {
-  }
+export class OnlineTrackerComponent {
 
-  public ngOnInit() {
-    let timer = Observable.timer(1000, 30000);
-    timer.subscribe((t) => this.updateOnlineStatus());
+  constructor(private onlineTrackerService: OnlineTrackerService) {
+    this.onlineTrackerService.updateOnlineStatus();
```

Comment:
You want to put this in the onInit function and reuse the helper  service I suggested above.